data/provider/src/main/java/tech/pegasys/teku/api/stateselector/StateSelector.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api.stateselector;
+
+import java.util.List;
+import java.util.Optional;
+import tech.pegasys.teku.datastructures.state.BeaconState;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+
+public interface StateSelector {
+  SafeFuture<List<BeaconState>> getState();
```

Comment:
Are there any cases where we need to get multiple states?  Following the pattern isn't bad but I suspect its very unlikely that we'd ever need a List here and could just have the Optional version as the only method for states (whereas blocks we do wind up selecting multiple sometimes).

Reply:
i cant think of any, i was just following the pattern initially, i'll make it just have the optional interface.